### PR TITLE
feat(#2281): Add OnVersioned to replace VersionMojo

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Add OnVersioned which will replace VersionMojo.
____
I don't think we need to perform such a heavy task with rewriting files in VersionMojo because we can solve the problem with object hashes in place where it is needed (DiscoveryMojo).
It could simplify the entire solution and speed up the build.

Closes: #2281

---

## PR-Codex overview
This PR focuses on improving the `VersionsMojo` class and introducing the `OnVersioned` class. 

### Detailed summary
- Remove the `VersionsMojo` class and replace it with the `OnVersioned` class.
- Introduce the `OnVersioned` class to parse raw strings into object names and hashes.
- Add tests for the `OnVersioned` class.
- Remove the catch block for `AccessDeniedException` in the `cleanUpClasses` method of `TranspileMojo`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon What do you think?

File: eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * If a version is not provided - behaves like {@link OnUnversioned}.
+ *
+ * @since 0.30
+ * @todo #2281:30min Implement OnVersioned class.
+ *  This class should parse raw string into object name and hash from object name with semver
+ *  version. In other words this class should replace the behavior of
+ *  {@link org.eolang.maven.VersionsMojo} class. When this class is implemented,
+ *  remove the {@link org.eolang.maven.VersionsMojo class}.
+ */
+public final class OnVersioned implements ObjectName {
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java:43]: @volodya-lombrozo I believe it's a good idea to add tests for the class, even disabled ones. So we would see how the object of the class is supposed to work

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java [hunk 2]
```diff
@@ -54,15 +56,19 @@ public final class DiscoverMojo extends SafeMojo {
     @Override
     public void exec() throws FileNotFoundException {
         final Collection<ForeignTojo> tojos = this.scopedTojos().notDiscovered();
-        final Collection<String> discovered = new HashSet<>(1);
+        final Collection<ObjectName> discovered = new HashSet<>(1);
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java:59]: @maxonfjvipon I strongly believe we should not check something by looking into console. We have the tests for that. If tests don't show errors - it seems that everything is fine.

File: eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java (hunk 1 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java:44]: @maxonfjvipon Agree. It is a good idea.

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Could you have a look one more time, please?

File: eo-maven-plugin/src/test/java/org/eolang/maven/name/OnVersionedTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        "9b883935257bd59d1ba36240f7e213d4890df7ca, org.eolang.tu#0.28.10",
+        "a7a4556bf1aa697324d805570f42d70affdddb75, org.eolang.wen#0.28.14",
+        "54d83d4b1d28075ee623d58fd742eaa529febd3d, org.eolang.th#0.28.2",
+        "6c6269d1f9a1c81ffe641538f119fe4e12706cb3, org.eolang.fri#0.28.4",
+        "9c9352890b5d30e1b89c9147e7c95a90c9b8709f, org.eolang.sat#0.28.5",
+        "17f89293e5ae6115e9a0234b754b22918c11c602, org.eolang.sun#0.28.6",
+        "5f82cc1edffad67bf4ba816610191403eb18af5d, org.eolang.penguin#0.28.7",
+        "be83d9adda4b7c9e670e625fe951c80f3ead4177, org.eolang.eagle#0.28.9"
+    })
+    @Disabled("Enable when OnVersioned.hash() is implemented")
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/java/org/eolang/maven/name/OnVersionedTest.java:90]: @volodya-lombrozo from the description of `OnVersioned`:
> This is object name that parses raw sting like:
> - "org.eolang.text#0.1.0" into "org.eolang.text" and "a1b2c3d"

I see narrow hash, but you have full hash in tests

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo check the build

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Could you have a look one more time, please?

File: eo-maven-plugin/src/test/java/org/eolang/maven/name/OnVersionedTest.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            name.hash().value(),
+            Matchers.equalTo(expected)
+        );
+    }
+
+    @Test
+    @Disabled("Enable when OnVersioned.toString() is implemented properly")
+    void convertsToStringWithNonEmptyVersion() {
+        final String string = "org.eolang.string#0.23.17";
+        final OnVersioned name = new OnVersioned(string);
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/java/org/eolang/maven/name/OnVersionedTest.java:104]: @volodya-lombrozo can we inline variable `string` here?

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Could you have a look one more time, please?

**Review by maxonfjvipon — APPROVED** [review]
@volodya-lombrozo LGTM! Thanks

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**